src/test/java/org/opensearch/geospatial/index/query/xypoint/XYPointQueryIT.java

```diff
@@ -0,0 +1,206 @@
+        String indexedShapePath = randomLowerCaseString();
+        createIndex(indexedShapeIndex, Settings.EMPTY, Map.of(indexedShapePath, XYShapeFieldMapper.CONTENT_TYPE));
+
+        final String shapeDocID = geospatialQueryTestHelperIT.indexDocumentUsingWKT(
+            indexedShapeIndex,
+            indexedShapePath,
+            "BBOX(-50, -40, -45, -55)"
+        );
+
+        final SearchResponse searchResponse = geospatialQueryTestHelperIT.searchUsingIndexedShapeIndex(
+            indexName,
+            indexedShapeIndex,
+            indexedShapePath,
+            shapeDocID,
+            xyPointFieldName
+        );
+        assertSearchResponse(searchResponse);
+        assertHitCount(searchResponse, 1);
+        MatcherAssert.assertThat(searchResponse.getHits().getAt(0).getId(), equalTo(secondDocumentID));
+
+        deleteIndex(indexName);
+        deleteIndex(indexedShapeIndex);
+    }
+
+    public void testIndexPointsPolygon() throws Exception {
+        createIndex(indexName, Settings.EMPTY, Map.of(xyPointFieldName, XYPointFieldMapper.CONTENT_TYPE));
+
+        final String firstDocumentID = geospatialQueryTestHelperIT.indexDocumentUsingWKT(indexName, xyPointFieldName, "POINT(-30 -30)");
+        geospatialQueryTestHelperIT.indexDocumentUsingWKT(indexName, xyPointFieldName, "POINT(-45 -50)");
+
+        double[] x = new double[] { -35, -35, -25, -25, -35 };
+        double[] y = new double[] { -35, -25, -25, -35, -35 };
+        LinearRing ring = new LinearRing(x, y);
+        Polygon polygon = new Polygon(ring);
+
+        final SearchResponse searchResponse = geospatialQueryTestHelperIT.searchUsingShapeRelation(
+            indexName,
+            xyPointFieldName,
+            polygon,
+            ShapeRelation.INTERSECTS
+        );
+        assertSearchResponse(searchResponse);
+        assertHitCount(searchResponse, 1);
+        MatcherAssert.assertThat(searchResponse.getHits().getAt(0).getId(), equalTo(firstDocumentID));
+
+        deleteIndex(indexName);
+    }
+
+    public void testIndexPointsMultiPolygon() throws Exception {
+        createIndex(indexName, Settings.EMPTY, Map.of(xyPointFieldName, XYPointFieldMapper.CONTENT_TYPE));
+
+        final String firstDocumentID = geospatialQueryTestHelperIT.indexDocumentUsingWKT(indexName, xyPointFieldName, "POINT(-30 -30)");
+        geospatialQueryTestHelperIT.indexDocumentUsingWKT(indexName, xyPointFieldName, "POINT(-40 -40)");
+        final String thirdDocumentId = geospatialQueryTestHelperIT.indexDocumentUsingWKT(indexName, xyPointFieldName, "POINT(-50 -50)");
+
+        LinearRing ring1 = new LinearRing(new double[] { -35, -35, -25, -25, -35 }, new double[] { -35, -25, -25, -35, -35 });
+        Polygon polygon1 = new Polygon(ring1);
+
+        LinearRing ring2 = new LinearRing(new double[] { -55, -55, -45, -45, -55 }, new double[] { -55, -45, -45, -55, -55 });
+        Polygon polygon2 = new Polygon(ring2);
```

Comment:
cut backlogs for creating these polygons via random generator. Doing hard coding is not the right way to do this.

Reply:
Yeah. We already have github issue #78 for that.